### PR TITLE
JAMES-2630 Add Reactor utils

### PR DESCRIPTION
**Pull request by blackheaven**

### CODE REVIEW

File: server/container/util/src/main/java/org/apache/james/util/ReactorUtils.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+import java.util.function.Supplier;
+
+import reactor.core.publisher.Mono;
+
+public class ReactorUtils {
+    public static <T> Mono<T> executeAndEmpty(Runnable runnable) {
+        return Mono.fromRunnable(runnable).then(Mono.empty());
+    }
+
+    public static <T> Function<Boolean, Mono<Boolean>> composeIfTrue(Supplier<Mono<T>> composeOperation) {
```
[the remark this answers is not in the document]
Comment (blackheaven) [reply server/container/util/src/main/java/org/apache/james/util/ReactorUtils.java]: Sure, that's supposed to replace `CompletableFutureUtil.composeIfTrue`

[the remark this answers is not in the document]
Comment (trantienduchn) [reply server/container/util/src/main/java/org/apache/james/util/ReactorUtils.java]: thanks for that explanation. I searched a place where we use `CompletableFutureUtil.composeIfTrue` and see that it resolve problem `CompletableFuture` can not do filter.
```java
    private CompletableFuture<Boolean> trySave(SimpleMailbox cassandraMailbox, CassandraId cassandraId) {
        return mailboxPathV2DAO.save(cassandraMailbox.generateAssociatedPath(), cassandraId)
            .thenCompose(CompletableFutureUtil.composeIfTrue(
                () -> retrieveMailbox(cassandraId)
                    .thenCompose(optional -> CompletableFuture
                        .allOf(optional
                                .map(storedMailbox -> mailboxPathV2DAO.delete(storedMailbox.generateAssociatedPath()))
                                .orElse(CompletableFuture.completedFuture(null)),
                            mailboxDAO.save(cassandraMailbox)))));
    }
```
With reactor, it is easy:
```java
    private CompletableFuture<Boolean> trySave(SimpleMailbox cassandraMailbox, CassandraId cassandraId) {
        return mailboxPathV2DAO.save(cassandraMailbox.generateAssociatedPath(), cassandraId)
            .filter(booleanResult -> booleanResult)
            .... // do something else;
    }
```
This Ultil method with me is not really needed now

[the remark this answers is not in the document]
Comment (blackheaven) [reply server/container/util/src/main/java/org/apache/james/util/ReactorUtils.java]: I'll Have a look at my refactorings, thanks

[the remark this answers is not in the document]
Comment (blackheaven) [reply server/container/util/src/main/java/org/apache/james/util/ReactorUtils.java]: Nice catch! thanks

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**